src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/net/CompositeSubscribeRequest.java

```diff
@@ -56,4 +58,17 @@ public SubscribeMessage getSubscribeMessage() {
   public StreamObserver<SubscribeResponse> getSubscribeResponseStream() {
     return subscribeResponseStream;
   }
+
+  @Override
+  public boolean equals(final Object o) {
```

Comment:
We should refrain from adding equals just for the purpose of testing. We should rather use a helper if Tests are the only ones that need it.